modules/unspents/src/dimensions.ts

```diff
@@ -290,434 +154,392 @@ Object.defineProperty(Dimensions.prototype, 'nInputs', {
-    case 'p2wsh':
-      return Dimensions.sum({ nP2wshInputs: 1 });
-    case 'p2tr':
-      switch (params.p2trSpendType) {
-        case 'keypath':
-          return Dimensions.sum({ nP2trKeypathInputs: 1 });
-        case 'scriptpath-level1':
-          return Dimensions.sum({ nP2trScriptPathLevel1Inputs: 1 });
-        case 'scriptpath-level2':
-          return Dimensions.sum({ nP2trScriptPathLevel2Inputs: 1 });
-        default:
-          throw new Error(`unsupported p2trSpendType: ${params.p2trSpendType}`);
-      }
+  /**
+   * Returns the dimensions of an output that will be created on a specific chain.
+   * Currently, this simply adds a default output.
+   *
+   * @param chain - Chain code as defined by utxolib.bitgo
+   * @return {Dimensions} - Dimensions for a single output on the given chain.
+   */
+  static fromOutputOnChain(chain: ChainCode): Dimensions {
+    return Dimensions.fromOutputScriptLength(Dimensions.getOutputScriptLengthForChain(chain));
   }
 
-  throw new Error(`unsupported chain ${chain}`);
-};
-
-/**
- * @param unspents
- * @return {Dimensions} sum of the dimensions for each unspent (@see Dimensions.fromUnspent())
- */
-Dimensions.fromUnspents = function (unspents) {
-  if (!Array.isArray(unspents)) {
-    throw new TypeError(`unspents must be array`);
-  }
-  // Convert the individual unspents into dimensions and sum them up
-  return Dimensions.sum(...unspents.map((u) => Dimensions.fromUnspent(u)));
-};
+  /**
+   * Return dimensions of an unspent according to `chain` parameter
+   * @param chain - Chain code as defined by utxo.chain
+   * @param params - Hint for unspents with variable input sizes (p2tr).
+   * @return {Dimensions} of the unspent
+   * @throws if the chain code is invalid or unsupported
+   */
+  static fromUnspent(
+    { chain }: { chain: number },
+    params: FromUnspentParams = { p2trSpendType: 'scriptpath-level1' }
+  ): Dimensions {
+    if (!isChainCode(chain)) {
+      throw new TypeError('invalid chain code');
+    }
 
-/**
- * @param transaction - bitcoin-like transaction
- * @param [param.assumeUnsigned] - default type for unsigned inputs
- * @return {Dimensions}
- */
-Dimensions.fromTransaction = function ({ ins, outs }, params) {
-  return Dimensions.fromInputs(ins, params).plus(Dimensions.fromOutputs(outs));
-};
+    switch (scriptTypeForChain(chain)) {
+      case 'p2sh':
+        return Dimensions.sum({ nP2shInputs: 1 });
+      case 'p2shP2wsh':
+        return Dimensions.sum({ nP2shP2wshInputs: 1 });
+      case 'p2wsh':
+        return Dimensions.sum({ nP2wshInputs: 1 });
+      case 'p2tr':
+        switch (params.p2trSpendType) {
+          case 'keypath':
+            return Dimensions.sum({ nP2trKeypathInputs: 1 });
+          case 'scriptpath-level1':
+            return Dimensions.sum({ nP2trScriptPathLevel1Inputs: 1 });
+          case 'scriptpath-level2':
+            return Dimensions.sum({ nP2trScriptPathLevel2Inputs: 1 });
```

Comment:
This and the `Dimensions.sum` above can also use the same constants/statics per my comment above, rather than recreating them each time `fromUnspent` is called.

Reply:
done